src/extension/module.js

```diff
@@ -1247,10 +1247,11 @@
               text: (titleI18n && titleI18n[language]) || title,
               action: () => {
                 if (url) {
-                  const baseTarget = url.startsWith('/')
-                    ? new URL(url, `https://${innerHost}/`)
-                    : url;
-                  const target = passReferrer ? `${baseTarget}?Referrer=${encodeURIComponent(location.href)}` : baseTarget;
+                  const target = url.startsWith('/') ? new URL(url, `https://${innerHost}/`) : new URL(url);
+                  if (passReferrer) {
+                    const { searchParams } = target;
+                    searchParams.append('referrer', encodeURIComponent(location.href));
```

Comment:
you don't need to encode the argument. the `searchParams` do this automatically.

Reply:
Should be fixed. Verified with a branch on milo.